src/D2L.Bmx/WriteHandler.cs

```diff
@@ -64,39 +69,50 @@ bool useCredentialProcess
 			using( File.Create( output ) ) { };
 		}
 
-		var data = parser.ReadFile( output );
+		string awsConfigFilePath = useCredentialProcess ? output : SharedCredentialsFile.DefaultConfigFilePath;
+		var awsConfig = GetParsedIniFileOrNull( awsConfigFilePath );
+
+		string awsCredentialsFilePath = useCredentialProcess ? SharedCredentialsFile.DefaultFilePath : output;
+		var awsCredentials = GetParsedIniFileOrNull( awsCredentialsFilePath );
```

Comment:
This is the main change.

My main pet peeve in this area is that `output` and `data` could have one of two different meanings, depending on the value of `useCredentialProcess`. I find this confusing lower down where the logic splits into two branches.

So, here, I changed the variable assignment so that `output` will no longer be used beyond this point, and we will have distinct names for the two different types of files/data.

As a nice side-effect, this is also the last place where we do any INI file parsing.